aspnetcore/fundamentals/error-handling.md

```diff
@@ -11,67 +11,307 @@ uid: fundamentals/error-handling
+
+## Exception-handling code
+
+Code in exception handling pages can throw exceptions. Production error pages should be tested and not contain complex code that may throw an exception.
+<!-- Review: original, which is not realistic 
+ > It's often a good idea for production error pages to consist of purely static content.
+
+ comments: - after you catch the exception, you need code to log the details and perhaps dynamically create a string with an error message. 
+-->
+
+### Response headers
+
+Once the headers for a response are sent:
+
+* The app can't change the response's status code.
+* Any exception pages or handlers can't run. The response must be completed or the connection aborted.
+
+## Server exception handling
+
+In addition to the exception handling logic in an app, the [HTTP server implementation](xref:fundamentals/servers/index) can handle some exceptions. If the server catches an exception before response headers are sent, the server sends a `500 - Internal Server Error` response without a response body. If the server catches an exception after response headers are sent, the server closes the connection. Requests that aren't handled by the app are handled by the server. Any exception that occurs when the server is handling the request is handled by the server's exception handling. The app's custom error pages, exception handling middleware, and filters don't affect this behavior.
+
+## Startup exception handling
+
+Only the hosting layer can handle exceptions that take place during app startup. The host can be configured to [capture startup errors](xref:fundamentals/host/web-host#capture-startup-errors) and [capture detailed errors](xref:fundamentals/host/web-host#detailed-errors).
+
+The hosting layer can show an error page for a captured startup error only if the error occurs after host address/port binding. If binding fails:
+
+* The hosting layer logs a critical exception.
+* The dotnet process crashes.
+* No error page is displayed when the HTTP server is [Kestrel](xref:fundamentals/servers/kestrel).
+
+When running on [IIS](/iis) (or Azure App Service) or [IIS Express](/iis/extensions/introduction-to-iis-express/iis-express-overview), a *502.5 - Process Failure* is returned by the [ASP.NET Core Module](xref:host-and-deploy/aspnet-core-module) if the process can't start. For more information, see <xref:test/troubleshoot-azure-iis>.
+
+## Database error page
+
+The Database developer page exception filter `AddDatabaseDeveloperPageExceptionFilter` captures database-related exceptions that can be resolved by using Entity Framework migrations. When these exceptions occur, an HTML response with details of possible actions to resolve the issue is generated. This page is enabled only in the Development environment. The following code was generated by the ASP.NET Core Razor Pages templates when individual user accounts were specified:
```

Comment:
> This page is enabled only in the Development environment.

I see what this means, but I think this should touch on the relationship between `UseDeveloperExceptionPage` and `AddDatabaseDeveloperPageExceptionFilter`.